aip/general/0008.md

```diff
@@ -7,22 +7,49 @@ placement:
   order: 40
 ---
 
-# AIP Style guide
+# AIP Style and Guidance
 
 AIP stands for **API Improvement Proposal**, which is a design document
-providing high-level, concise documentation for API development. The goal is for
-these documents to serve as the source of truth for API-related documentation at
-Google and the way API teams discuss and come to consensus on API guidance.
+providing high-level, concise documentation for API design and development. The
+goal is for these documents to serve as the source of truth for API-related
+documentation at Google and the way API teams discuss and come to consensus on
+API guidance.
 
 AIPs are most useful when they are clear and concise, and cover a single topic
 or inquiry well. In the same way that AIPs describe consistent patterns and
 style for use in APIs, they also _follow_ consistent patterns and style.
 
 ## Guidance
 
-AIPs **must** cover a single, discrete topic, and **should** fundamentally
-answer the question, "What do I do?" with actionable guidance. AIPs **may** also
-cover what _not_ to do, but **should not** cover _only_ anti-patterns.
+- AIPs **must** cover a single, discrete topic, and provide clear, actionable
+  guidance.
+- AIPs **must not** duplicate or contradict guidance in another AIP.
+- AIPs **may** also cover what _not_ to do, but **should not** cover _only_
+  anti-patterns.
+- If AIP guidance is conditional (e.g. a design pattern such as Jobs), the
+  guidance **must** clearly explain under what conditions the guidance should be
+  followed.
+
+AIPs **must** contain API guidance that provide a clear benefit to authors of
+one or more types of clients or their authors, including but not limited to:
+
+- Asset inventories which can be used to audit and analyze resources.
+- Command line interfaces for exploration and simple automation.
+- Custom controllers (e.g. auto-scalers) which poll live state and adjust
+  resource configruation accordingly.
+- [IaC][] clients for orchestration and automation of multiple resources.
+- Reccomendation tools which provide guidance on which APIs are useful for
+  specific use cases, and how to use them.
+- SDKs to interact with an API from a programming language, often used heavily
+  for data-plane operations.
+- Security orchestration, automation, and remediation tools.
+- Simple scripts to automate or orchestrate tasks.
+- Test frameworks.
+- Tools that operate on resource data at rest.
+- Visual User Interfaces for visualization and one-off manual actions.
+- Users.
+
+AIP guidance **must not** be a signifcant detriment to a client's usability.
```

Comment:
Typo: signifcant

Reply:
Also, this mentions usability, but we also care about implementation difficulty.

Reply:
ack, added implementation difficulty.